test/runtests.jl

```diff
@@ -169,6 +171,14 @@ SOFTWARE.
         @test frule_count == 16
         @test norm(g - I) < 1e-6
     end
+    @testset "exp!" begin
+        # Gradient of trace
+        fexp = x -> sum(LinearAlgebra.exp!(copy(x)))
```

Comment:
Did you mean to use the trace here instead of sum?

Probably also worth testing that the `frule` is being used under the hood. Perhaps use a wrapper function around `exp!` that tests that the element type of the input is not `ForwardDiff.Dual`? You will need to define an `frule` for that function which just calls `exp!`'s `frule`.

Reply:
No, I just wanted a scalar so sum is fine. The rule is called because without it the correctness test fails: #14

Reply:
I see. The comment above the line of code says `# Gradient of trace` that's why I asked.

Reply:
Does ForwardDiff on its own without this package error or give incorrect gradient?

Reply:
Ah, a copy paste mistake 😅
Without this rule there is no method matching `exp!` for dual matrices, and without the fix in the pr defining that method with this package returns an incorrect answer.